space-time.Rmd

```diff
@@ -138,9 +116,42 @@ ggplot(mi_cities, aes(lon, lat)) +
   coord_quickmap()
 ```
 
-### Raster images
+### Adding area metadata
 
-Instead of displaying context with vector boundaries, you might want to draw a traditional map underneath. This is called a raster image. The easiest way to get a raster map of a given area is to use the ggmap package, which allows you to get data from a variety of online mapping sources including OpenStreetMap and Google Maps. Downloading the raster data is often time consuming so it's a good idea to cache it in a rds file. \index{ggmap} \index{Raster data}
+Sometimes metadata is associated not with a point, but with an area. For example, we can create `mi_census` which provides census information about each county in MI:
+
+```{r}
+mi_census <- midwest %>%
+  tbl_df() %>%
+  filter(state == "MI") %>% 
+  mutate(county = tolower(county)) %>%
+  select(county, area, poptotal, percwhite, percblack)
+mi_census
+```
+
+We can't map this data directly because it has no spatial component. Instead, we must first join it to the vector boundaries data. This is not particularly space efficient, but it makes it easy to see exactly what data is being plotted. Here I use `dplyr::left_join()` to combine the two datasets and create a choropleth map. \index{Choropleth}
+
+```{r}
+census_counties <- left_join(mi_census, mi_counties, by = c("county" = "id"))
+census_counties
+
+ggplot(census_counties, aes(lon, lat, group = county)) + 
+  geom_polygon(aes(fill = poptotal)) + 
+  coord_quickmap()
+
+ggplot(census_counties, aes(lon, lat, group = county)) + 
+  geom_polygon(aes(fill = percwhite)) + 
+  coord_quickmap()
+```
+
+
+## Raster maps {#rastermaps}
+
+Instead of displaying context with vector boundaries, you might want to draw a traditional map underneath. This is called a raster image. The easiest way to get a raster map of a given area is to use the ggmap package, which allows you to get data from a variety of online mapping sources including OpenStreetMap and Google Maps. Downloading the raster data is often time consuming so it's a good idea to cache it in a rds file, as illustrated here: \index{ggmap} \index{Raster data}
```

Comment:
Is this still true now that Google maps requires an api key? Is there anything newer that's easier to use? (Perhaps based on [ceramic](https://github.com/hypertidy/ceramic)? Maybe we could help make [this](https://github.com/hypertidy/ceramic/issues/4) happen?)